chat/src/test/java/net/md_5/bungee/api/chat/ComponentsTest.java

```diff
@@ -11,24 +11,120 @@
 public class ComponentsTest
 {
 
+    @Test
+    public void testEmptyComponentBuilder()
+    {
+        ComponentBuilder builder = new ComponentBuilder();
+
+        BaseComponent[] parts = builder.create();
+        Assert.assertEquals( parts.length, 0 );
+
+        for ( int i = 0; i < 3; i++ )
+        {
+            builder.append( "part:" + i );
+            parts = builder.create();
+            Assert.assertEquals( parts.length, i + 1 );
+        }
+    }
+
+    @Test
+    public void testDummyRetaining()
+    {
+        ComponentBuilder builder = new ComponentBuilder();
+        Assert.assertNotNull( builder.getCurrentComponent() );
+        builder.color( ChatColor.GREEN );
+        builder.append( "test ", ComponentBuilder.FormatRetention.ALL );
+        Assert.assertEquals( builder.getCurrentComponent().getColor(), ChatColor.GREEN );
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void testComponentGettingExceptions()
+    {
+        ComponentBuilder builder = new ComponentBuilder();
+        builder.getComponent( -1 );
```

Comment:
Test fails here, you should rewrite it to few methods.

Reply:
The test passes for me, it should expect for IndexOutOfBoundsException